### DISCUSSION TITLE
Nice app - Site down?

### DISCUSSION THREAD

**Discussion by jamestagal** [itswadesh/svelte-commerce#87] [General]
G'day there,

I just came across this repo and it looks great. Would be cool to be able to define products clientside and have this e-commerce app scrape the site like go-commerce was doing. Is that how ```Svelte-commerce``` works?

BTW. I am seeing errors from your site ```https://arialshop.com/```

<img width="1085" alt="Screen Shot 2024-01-08 at 9 45 46 am" src="https://github.com/itswadesh/svelte-commerce/assets/6309595/522a45b0-7084-4df5-a431-4a662b9d6490">

**Comment by itswadesh** [comment]
Hi, thanks for letting us know. Site is up now.

I am not sure how go-commerce works and it would be beneficial to have products client side. Can you please send me some more reference.

**Reply by jamestagal** [reply]
Hi @itswadesh 

Great thanks.

I am building sites with https://plenti.co/ Plenti is an open source Build-Time Render (BTR) Engine. Some people refer to it as a Static Site Generator (SSG). It uses Svelte on the frontend and Go for the backend.  Also it has Git-backed CMS and it also uses a simple JSON data source instead of Markdown . So products would be located client side in the ```.json``` data source. I'm looking for an ecommerce solutions for when I need to build a simple store. 

My comment on go-commerce was just that it would be cool to be able to do secure product checkouts completely client side. This is normally hard because the client can be manipulated by the user, but if the backend service looks to the endpoint and verifies the pricing metadata itself, it's possible. I haven't really looked at your project and tried anything. yet.

I see you have support for Shopify but do also you support other payment getaways such as Square or Stripe? 

-Ben

**Reply by itswadesh** [reply]
Hi @jamestagal 

The idea of Plenti looks interesting. I am not sure if payment gateway integration can be done using Plenti `Go` based backend? If not then we can use serverless functions for the same.

Yes we can use simple JSON based data source instead of a full fledge backend.

Yes svelte-commerce supports Shopify as well as other payment gateways like Stripe. Square integration is not started yet, but possible.

Regards
Swadesh

**Reply by jamestagal** [reply]
Hi @itswadesh 

Thanks for your reply. That's great. I think serverless functions sound like a good approach but I'm not very familiar with this technology. 
It sounds like serverless functions would need to be on a hosted service ? I deploy to Cloudflare and I think If the adaptor is Clouldflare then endpoints will be Cloudflare workers deployed to edge and also use the built-in Clouldflare KV store directly instead of database but I'm not sure? 

Ben

**Reply by itswadesh** [reply]
In that case Nitro (https://nitro.unjs.io/deploy/workers/) could be a good option
